sycl/include/CL/sycl/pipes.hpp

```diff
@@ -12,11 +12,33 @@
 #include <CL/__spirv/spirv_types.hpp>
 #include <CL/sycl/stl.hpp>
 
+#ifndef __SYCL_DEVICE_ONLY__
+#include <iostream>
+#endif // __SYCL_DEVICE_ONLY__
+
 namespace cl {
 namespace sycl {
 
-template <class name, class dataT, int32_t min_capacity = 0> class pipe {
+// Utility to dump typeid
+template <typename T> class wrap {};
+template <typename Type>
+std::string __str_type_name() {
+  return std::string(typeid(wrap<Type>).name());
+}
+
+template <typename PipeName>
+inline void __pipe_printer(std::string Input, int32_t S, int32_t A, int32_t C) {
```

Comment:
Probably these 2 functions should be guarded by `#ifndef __SYCL_DEVICE_ONLY__` too.

Reply:
Well, I had some plans on __str_type_name() in a close future ;) But I agree, that __pipe_printer() should be guarded.